test/lib/test-index.ts

```diff
@@ -0,0 +1,58 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import assert = require('assert');
+import http = require('http');
+import mockery = require('mockery');
+import _ = require('lodash');
+import MockBLPAPI = require('./mock-blpapi');
+import sio = require('socket.io');
+
+// Setup mock of blpapi
+mockery.registerMock('blpapi', MockBLPAPI);
+mockery.enable({
+    useCleanCache: true,
+    warnOnReplace: true,
+    warnOnUnregistered: false
+});
+
+// Setup an Socket.IO server for IPC need
+var s: http.Server = http.createServer();
+s.listen(3333);
+s.on('listening', (): void => {
+    console.log('ipc server listening on port 3333');
+});
+var io: SocketIO.Server = sio(s);
+io.on('connection', (socket: SocketIO.Socket): void => {
+    console.log('ipc channel connected');
+    MockBLPAPI.ipc = socket;
+    socket.on('set-instructions', (instructions: MockBLPAPI.IInstruction, cb: Function): void => {
+        _.forEach(instructions, (i: any, key: string): void => {
+            MockBLPAPI.instructions[key] = i;
+        });
+        cb();
+    });
+    socket.on('clear-instructions', (arg: string[], cb: Function): void => {
+        if (arg) {
+            _.forEach(arg, (s: string): void => {
+                delete MockBLPAPI.instructions[s];
+            });
+        } else {
+            MockBLPAPI.instructions = {};
+        }
+        cb();
+    });
+});
+
+process.on('exit', (): void => {
+    mockery.deregisterAll();
+    mockery.disable();
+    MockBLPAPI.ipc.disconnect(true);
+});
+
+// Run actual server.ts
+import server = require('../../lib/server');
```

Comment:
I leave importing the server here intentionally because we need to setup the `mockery` module to do dependency injection first. Import the server beforehand will nullify the mock layer.